onnx/test/shape_inference_test.py

```diff
@@ -155,11 +170,20 @@ def _assert_inferred(
         )
         names_in_vis = {x.name for x in vis}
         vis = [x for x in graph.value_info if x.name not in names_in_vis] + vis
+        shape_inference_function_arg_name = "has_inference_function"
+        has_inference_func = True
+        if shape_inference_function_arg_name in kwargs:
+            if not kwargs[shape_inference_function_arg_name]:
+                has_inference_func = False
+            kwargs.pop(shape_inference_function_arg_name)
         inferred_model = self._inferred(graph_or_model, **kwargs)
         inferred_vis = list(inferred_model.graph.value_info)
         vis = sorted(vis, key=lambda x: x.name)  # type: ignore[no-any-return]
         inferred_vis = sorted(inferred_vis, key=lambda x: x.name)  # type: ignore
-        assert len(vis) == len(inferred_vis)
+        if has_inference_func:
```

Comment:
I would add a comment to explain the logic here. It is not immediately clear what the +1 is doing or what invariants it is keeping?